data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/AvroConsumer.java

```diff
@@ -0,0 +1,103 @@
+    private static final Logger LOG = LoggerFactory.getLogger(AvroConsumer.class);
+    private Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>();
+    private long lastReadOffset = 0L;
+    private KafkaConsumer<String, GenericRecord> kafkaAvroConsumer;
+    private volatile long lastCommitTime = System.currentTimeMillis();
+
+    @Override
+    public void consumeRecords(final KafkaConsumer<String, GenericRecord> consumer, final AtomicBoolean status,
+                               final Buffer<Record<Object>> buffer, final TopicsConfig topicConfig,
+                               final KafkaSourceConfig kafkaSourceConfig, PluginMetrics pluginMetrics,
+                               final String schemaType) {
+        KafkaSourceBufferAccumulator kafkaSourceBufferAccumulator = new KafkaSourceBufferAccumulator(topicConfig,kafkaSourceConfig,
+                schemaType,pluginMetrics);
+        kafkaAvroConsumer = consumer;
+        try {
+            consumer.subscribe(Arrays.asList(topicConfig.getName()));
+            while (!status.get()) {
+                offsetsToCommit.clear();
+                ConsumerRecords<String, GenericRecord> records = consumer.poll(Duration.ofMillis(100));
+                if (!records.isEmpty() && records.count() > 0) {
+                    for (TopicPartition partition : records.partitions()) {
+                        List<Record<Object>> kafkaRecords = new ArrayList<>();
+                        List<ConsumerRecord<String, GenericRecord>> partitionRecords = records.records(partition);
+                        for (ConsumerRecord<String, GenericRecord> consumerRecord : partitionRecords) {
+                            offsetsToCommit.put(new TopicPartition(consumerRecord.topic(), consumerRecord.partition()),
+                                    new OffsetAndMetadata(consumerRecord.offset() + 1, null));
+                            kafkaRecords.add(kafkaSourceBufferAccumulator.getEventRecord(consumerRecord.value().toString()));
+                            lastReadOffset = partitionRecords.get(partitionRecords.size() - 1).offset();
+                        }
+                        if (!kafkaRecords.isEmpty()) {
+                            kafkaSourceBufferAccumulator.writeAllRecordToBuffer(kafkaRecords, buffer, topicConfig);
+                        }
+                    }
+                    if (!offsetsToCommit.isEmpty() && topicConfig.getAutoCommit().equalsIgnoreCase("false")) {
+                        lastCommitTime = kafkaSourceBufferAccumulator.commitOffsets(consumer, lastCommitTime, offsetsToCommit);
+                    }
+                }
+            }
+        } catch (Exception exp) {
+            LOG.error("Error while reading avro records from the topic...", exp);
+        }
+    }
+
+    @Override
+    public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+        LOG.trace("onPartitionsAssigned() callback triggered and Closing the consumer...");
```

Comment:
We need to revisit logging to make sure we don't flood logs with too many logs/

Reply:
Resolved. removed the unwanted logs